drivers/tmp00x/tmp00x.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (C) 2014 PHYTEC Messtechnik GmbH
+ *               2017 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     drivers_tmp00x
+ * @{
+ *
+ * @file
+ * @brief       Driver for the TI TMP00X (TMP006 and TMP007) Infrared Thermopile Sensor.
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author      Sebastian Meiling <s@mlng.net>
+ * @author      Jannes Volkens <jannes.volkens@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <math.h>
+
+#include "log.h"
+#include "periph/i2c.h"
+#include "tmp00x.h"
+#include "tmp00x_regs.h"
+#if TMP00X_USE_LOW_POWER
+#include "xtimer.h"
+#endif
+
+#define ENABLE_DEBUG                (0)
+#include "debug.h"
+
+#define TMP00X_CONFIG_RST           (1 << 15)
+
+#define TMP00X_CONFIG_MOD_SHIFT     (12U)
+#define TMP00X_CONFIG_MOD_MASK      (0x7000)
+#define TMP00X_CONFIG_MOD(x)        (((uint16_t)(((uint16_t)(x)) << TMP00X_CONFIG_MOD_SHIFT))\
+                                     & TMP00X_CONFIG_MOD_MASK)
+#define TMP00X_CONFIG_MOD_CC        (0x07)
+#define TMP00X_CONFIG_MOD_OFF       (0x00)
+
+#define TMP00X_CONFIG_CR_SHIFT      (9U)
+#define TMP00X_CONFIG_CR_MASK       (0x0E00)
+#define TMP00X_CONFIG_CR(x)         (((uint16_t)(((uint16_t)(x)) << TMP00X_CONFIG_CR_SHIFT))\
+                                     & TMP00X_CONFIG_CR_MASK)
+
+#define TMP00X_CONFIG_DRDY_PIN_EN   (1 << 8)
+
+#if defined (MODULE_TMP006)
+#define TMP006_CONFIG_DRDY          (1 << 7)
+#define TMP006_DID_VALUE            (0x0067) /**< Device ID */
+#else
+#define TMP007_CONFIG_DRDY          (1 << 6)
+#define TMP007_DID_VALUE            (0x0078) /**< Device ID */
+#endif
+
+#define TMP00X_MID_VALUE            (0x5449) /**< Manufacturer ID */
+
+#define BUS                         (dev->p.i2c)
+#define ADDR                        (dev->p.addr)
+
+int tmp00x_init(tmp00x_t *dev, const tmp00x_params_t *params)
+{
+    /* check parameters */
+    assert(dev && params);
+
+    uint8_t reg[2];
+    uint16_t tmp;
+
+    /* initialize the device descriptor */
+    dev->p = *params;
+
+    if (dev->p.rate > TMP00X_CONFIG_CR_AS16) {
+        LOG_ERROR("tmp00x_init: invalid conversion rate!\n");
+        return -TMP00X_ERROR_CONF;
+    }
+
+    /* test device id */
+    i2c_acquire(BUS);
+    #if defined (MODULE_TMP006)
+    if (i2c_read_regs(BUS, ADDR, TMP006_REGS_DEVICE_ID, reg, 2, 0) < 0) {
```

Comment:
If you replace the `TMP006_REGS_DEVICE_ID` with `TMP00x_REGS_DEVICE_ID` then you don't need to `ifdef` this.

So in the header you can just go
```c
#ifdef MODULE_TMP006
#define TMP006_REGS_DEVICE_ID ???
...
#define TMP00X_REGS_DEVICE_ID TMP006_REGS_DEVICE_ID

#elif define(MODULE_TMP007)
#define TMP007_REGS_DEVICE_ID ???
...
#define TMP00X_REGS_DEVICE_ID TMP007_REGS_DEVICE_ID
#else
#error "TMP00X DRIVER not selected or supported"
#endif
```

Also all preprocessor directives should not have any leading spaces
so
```c
    #ifdef NOT_LIKE_THIS
#elif define(THIS_IS_BETTER)
#endif
```